src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -46,6 +46,22 @@ internal sealed class VariableState
             }
         }
 
+        /// <summary>
+        /// Represents the result of visiting an expression.
+        /// Contains a result type which tells us whether the expression may be null,
+        /// and an l-value type which tells us whether we can assign null to the expression.
+        /// </summary>
+        private struct VisitResult
+        {
+            public TypeSymbolWithAnnotations ResultType;
+            private TypeSymbolWithAnnotations _lValueType;
+            public TypeSymbolWithAnnotations LValueType
```

Comment:
How do we represent the case where it is not an lvalue at all?

Reply:
In such case, `LValueType` gets the same value as `ResultType`. This emulates previous behavior, where `VisitLvalue` would default to calling `VisitRValue`.

---
In reply to: [258140214](https://github.com/dotnet/roslyn/pull/33447#discussion_r258140214) [](ancestors = 258140214)